src/cascadia/TerminalApp/ColorScheme.cpp

```diff
@@ -120,36 +131,60 @@ void ColorScheme::LayerJson(const Json::Value& json)
     }
 }
 
-winrt::hstring ColorScheme::Name() const noexcept
-{
-    return _schemeName;
-}
-
-winrt::com_array<Color> ColorScheme::Table() const noexcept
+// Method Description:
+// - Create a new serialized JsonObject from an instance of this class
+// Arguments:
+// - scheme: a ColorScheme object that will be converted into a serialized JsonObject
+// Return Value:
+// - a serialized JsonObject from the values in scheme
+Json::Value ColorScheme::ToJson(const TerminalApp::ColorScheme& scheme)
 {
-    winrt::com_array<Color> result{ COLOR_TABLE_SIZE };
-    std::transform(_table.begin(), _table.end(), result.begin(), [](til::color c) -> Color { return c; });
-    return result;
+    Json::Value json{ Json::ValueType::objectValue };
+    const auto schemeImpl{ winrt::get_self<implementation::ColorScheme>(scheme) };
+    schemeImpl->UpdateJson(json);
+    return json;
 }
 
-Color ColorScheme::Foreground() const noexcept
+// Method Description:
+// - Update the given json object with values from this object.
+// Arguments:
+// - json: an object which will be a serialization of a ColorScheme object.
+// Return Value:
+// <none>
+void ColorScheme::UpdateJson(Json::Value& json)
 {
-    return _defaultForeground;
-}
+    JsonUtils::SetValueForKey(json, NameKey, _Name);
+    JsonUtils::SetValueForKey(json, ForegroundKey, _Foreground);
+    JsonUtils::SetValueForKey(json, BackgroundKey, _Background);
+    JsonUtils::SetValueForKey(json, SelectionBackgroundKey, _SelectionBackground);
+    JsonUtils::SetValueForKey(json, CursorColorKey, _CursorColor);
 
-Color ColorScheme::Background() const noexcept
-{
-    return _defaultBackground;
+    int i = 0;
+    for (const auto& current : TableColors)
+    {
+        JsonUtils::SetValueForKey(json, current, _table.at(i));
+        i++;
+    }

```

Comment:
this might be the one time where going an indexed for loop is better.

```suggestion
    for (unsigned int i = 0; i < TableColors.size(); ++i)
    {
        JsonUtils::SetValueForKey(json, til::at(TableColors, i), _table.at(i));
    }
```

If you find yourself wishing you had an _index_ when you're iterating over something with _iterators_.. sometimes you can just use a plain old loopy boi